lib/services/app.go

```diff
@@ -158,3 +158,174 @@ func UnmarshalAppServer(data []byte, opts ...MarshalOption) (types.AppServer, er
 	}
 	return nil, trace.BadParameter("unsupported app server resource version %q", h.Version)
 }
+
+// CompareAppServers returns true if two application servers are equal.
+func CompareAppServers(as1, as2 types.AppServer) bool {
```

Comment:
You might need to update it to ignore non-significant fields for AppServer and other resources you need but otherwise seems better to use that than having separate CompareXXX methods? Esp. because if we add new fields to them, we'll have to remember to update these comparison methods.

Reply:
Looks like `CompareResources` works fine, so I'll go ahead and close this.